pytorch_lightning/__init__.py

```diff
@@ -37,10 +38,11 @@
 - https://pytorch-lightning.readthedocs.io/en/latest
 - https://pytorch-lightning.readthedocs.io/en/stable
 """
-
-_logger = python_logging.getLogger("lightning")
-_logger.addHandler(python_logging.StreamHandler())
-_logger.setLevel(python_logging.INFO)
+python_logging.basicConfig(
+    stream=sys.stderr,
+    format="%(message)s",
+    level=python_logging.INFO,
+)
```

Comment:
this is problematic because it would enable logs for other libraries that import pytorch_lightning

Reply:
yes, and if they properly set their logging level, everything should be fine and if they don't want to, they can filter it out or overwrite it.

Reply:
The proper solution is still to correctly configure the lightning logger handlers. A library should not run `basicConfig` IMO because it impacts all logging (inside our outside)

Reply:
@carmocca can you please show me how?

Reply:
An alternative to calling basicConfig:

```python
_root_logger = python_logging.getLogger()
_formatter = python_logging.Formatter("%(message)s")
_stream_handler = python_logging.StreamHandler(stream=sys.stderr)
_stream_handler.setFormatter(_formatter)
_root_logger.addHandler(_stream_handler)
_root_logger.setLevel(python_logging.INFO)
```

Is this what you mean?

Reply:
Yes, you'd be missing the `setFormatter` and also `getLogger('lightning')`

If that doesn't work, I can have a look

Reply:
For example, how I do it in my lib:

https://github.com/jpuigcerver/PyLaia/blob/5e0d4ca68f236eeafd8578130ec914750e41c9aa/laia/common/logging.py#L127-L158

You might also want to reset the handlers, otherwise you might add duplicates:

https://github.com/jpuigcerver/PyLaia/blob/5e0d4ca68f236eeafd8578130ec914750e41c9aa/laia/common/logging.py#L86-L88

Reply:
ok, I'm a bit overwhelmed by this now. There doesn't seem to be a agreement of how logging should be configured in general. Everywhere I go I read some different approach. 
I initially thought that libraries shouldn't configure logging for the user. However, in Lightning we need to provide a good default because we can't expect the user to call basicConfig or setup the root logger, they want to see the output right away.

Could you give me some more guidance?
Current approach in this PR is: 1) integrate logs in Lightning in the default logging hierarchy, organized by modules. 
2) Configure logging at the top level on the root logger as a good default 3) User can override these settings either at the root level or further down 4) Do not add any stream handlers in between except on the root logger, otherwise we get duplicate logs.

Given your previous message, should I do something else with the handlers?

Reply:
```
_root_logger = logging.getLogger("lightning")
_root_handler = logging.StreamHandler(sys.stdout)
_root_handler.setFormatter(fmt)
_root_logger.addHandler(root_handler)
_root_logger.setLevel(logging.INFO)
```
I think you want something like this? (just adding a formatter to the stream handler)

Reply:
Yes, I tried this and it works:

```python
# Configure logging
_lightning_handler = python_logging.StreamHandler(sys.stderr)
_lightning_handler.setFormatter(python_logging.Formatter("%(message)s"))
_lightning_handler.setLevel(python_logging.INFO)

_lightning_logger = python_logging.getLogger("pytorch_lightning")
_lightning_logger.handlers = []
_lightning_logger.addHandler(_lightning_handler)
_lightning_logger.setLevel(python_logging.INFO)
_lightning_logger.propagate = 0
```

Note that for your snippet:

```python
import pytorch_lightning as pl
import logging

logging.info("test")
pl.seed_everything(1234)
```

`test` **should not** be logged, as it is a responsability of the user to configure their logging.

whereas it **should** be logged if the user did:

```python
import pytorch_lightning as pl
import logging as log
logging = log.getLogger("pytorch_lightning")

logging.info('I should be logged')
pl.seed_everything(1234)
```

because the user would be relying on our (already configured) logger

Reply:
> test should not be logged

Why not if you set default level to INFO?

Reply:
thanks for investigating @carmocca 
setting propagate=False, how would you address the test cases, for example tests/test_profiler if messages don't propagate to the root?

Reply:
because the user did not set the default level to INFO

we set it for **our** library logger

Reply:
Also, should we consider relying on third party library for this: https://github.com/Delgan/loguru

<img width="999" alt="Screenshot 2021-01-24 at 10 10 30" src="https://user-images.githubusercontent.com/12861981/105627078-67530c80-5e2c-11eb-984b-1cb48a9d67ef.png">